manim/utils/iterables.py

```diff
@@ -53,24 +71,24 @@ def adjacent_pairs(objects: Sequence) -> zip:
 
     Examples
     --------
-    Normal usage::
+    .. code-block:: pycon
 
-        list(adjacent_pairs([1, 2, 3, 4]))
-        # returns [(1, 2), (2, 3), (3, 4), (4, 1)]
+        >>> list(adjacent_pairs([1, 2, 3, 4]))
+        [(1, 2), (2, 3), (3, 4), (4, 1)]
     """
     return adjacent_n_tuples(objects, 2)
 
 
-def all_elements_are_instances(iterable: Iterable, Class) -> bool:
+def all_elements_are_instances(iterable: Iterable[object], Class: type[object]) -> bool:
     """Returns ``True`` if all elements of iterable are instances of Class.
     False otherwise.
     """
     return all(isinstance(e, Class) for e in iterable)
 
 
 def batch_by_property(
-    items: Sequence, property_func: Callable
-) -> list[tuple[list, Any]]:
+    items: Sequence[T], property_func: Callable[[T], U]
+) -> list[tuple[list[T], U]]:
```

Comment:
```suggestion
    items: Iterable[T], property_func: Callable[[T], U]
) -> list[tuple[list[T], U]]:
```

I think it would also be great to deprecate this function, it is equivalent to [`itertools.groupby`](https://docs.python.org/3/library/itertools.html#itertools.groupby) (but way slower as pure Python). In this PR or a separate one?

Reply:
I would prefer a separate PR as to leave this PR just for type hints